### PR TITLE
Add source code for Chapter 14

### PR DESCRIPTION
**Pull request by Peter-VanWeert**

### CODE REVIEW

File: Exercises/NoModules/Chapter 14/Soln14_01/Animals.cpp [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+// Constructor
+Animal::Animal(std::string_view name, int weight) 
+	: m_name{name}, m_weight{weight}
+{}
+
+// Identify the animal
+void Animal::who() const
+{
+  std::cout << "My name is " << m_name << " and I weigh " << m_weight << "lbs." << std::endl;
```
Comment (MarcGregoire) [inline Exercises/NoModules/Chapter 14/Soln14_01/Animals.cpp:14]: format?

File: Exercises/NoModules/Chapter 14/Soln14_04/Person.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  // you could validate the arguments by doing reasonableness checks. 
+  // e.g. Name mustn't be empty, and age should be less than 130 say.
+  // To handle a failure sensibly we really need exceptions, 
+  // but we don't get to those until chapter 16. 
+}
+
+// Display details of Person object
+void Person::who() const 
+{
+  std::cout << "\nThis is " << m_name << " who is " << m_age << " years old." << std::endl;
```
Comment (MarcGregoire) [inline Exercises/NoModules/Chapter 14/Soln14_04/Person.cpp:20]: format?

File: Exercises/NoModules/Chapter 14/Soln14_04/Person.cpp [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+std::string_view Person::getGenderString() const
+{
+  switch (m_gender)
+  {
+  case Gender::male:   return "a male";
+  case Gender::female: return "a female";
+  case Gender::other:  return "an other-gendered";
+  }
+
+  // Unreacheble return statement required by some compilers
```
Comment (MarcGregoire) [inline Exercises/NoModules/Chapter 14/Soln14_04/Person.cpp:37]: Unreachable

**Review by MarcGregoire — CHANGES_REQUESTED** [review]

File: Examples/NoModules/Chapter 14/Ex14_01/Ex14_01A.cpp [hunk 4]
```diff
@@ -0,0 +1,27 @@
+// Changing the access specification of inherited members
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/NoModules/Chapter 14/Ex14_01/Ex14_01A.cpp:1]: Yeah, Ex14_01 and Ex14_01A seem to have been somewhat mixed...

File: Examples/NoModules/Chapter 14/Ex14_01/Carton.h [hunk 5]
```diff
@@ -0,0 +1,18 @@
+// Carton.h - defines the Carton class with the Box class as (private) base
+#ifndef CARTON_H
+#define CARTON_H
+
+#include <string>
+#include <string_view>
+#include "Box.h"
+
+class Carton : private Box
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/NoModules/Chapter 14/Ex14_01/Carton.h:9]: Yeah, Ex14_01 and Ex14_01A seem to have been somewhat mixed...

File: Examples/NoModules/Chapter 14/Ex14_01/Ex14_01A.cpp [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  std::cout << "carton occupies " << sizeof carton << " bytes" << std::endl;
+  std::cout << "candyCarton occupies " << sizeof chocolateCarton << " bytes" << std::endl;
+
+  // Now volumes...
+  std::cout << "box volume is " << box.volume() << std::endl;
+  std::cout << "carton volume is " << carton.volume() << std::endl;
+  std::cout << "chocolateCarton volume is " << chocolateCarton.volume() << std::endl;
+
+  // Uncomment any of the following for an error...
+  // std::cout << "chocolateCarton length is " << chocolateCarton.getLength() << std::endl;
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/NoModules/Chapter 14/Ex14_01/Ex14_01A.cpp:23]: Yeah, Ex14_01 and Ex14_01A seem to have been somewhat mixed...

File: Examples/NoModules/Chapter 14/Ex14_01A/Box.h [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+#define BOX_H
+#include <iostream>                         // For standard streams
+#include <iomanip>                          // For stream manipulators
+
+class Box
+{
+private:
+  double length {1.0};
+  double width {1.0};
+  double height {1.0};
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/NoModules/Chapter 14/Ex14_01A/Box.h:12]: Old code...

File: Examples/NoModules/Chapter 14/Ex14_01A/Box.h [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+public:
+  // Constructors
+  Box(double lv, double wv, double hv) : length {lv}, width {wv}, height {hv} {}
+  Box() = default;                          // No-arg constructor
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/NoModules/Chapter 14/Ex14_01A/Box.h:17]: Yeah, Ex14_01 and Ex14_01A seem to have been somewhat mixed into the Ex14_01 folder, and the Ex14_01A one is just old code...

File: Examples/NoModules/Chapter 14/Ex14_01A/Box.h [hunk 9]
```diff
@@ -0,0 +1,41 @@
+// Box.h - defines Box class
+#ifndef BOX_H
+#define BOX_H
+#include <iostream>                         // For standard streams
+#include <iomanip>                          // For stream manipulators
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/NoModules/Chapter 14/Ex14_01A/Box.h:5]: Yeah, Ex14_01A is just a folder with old code...

File: Examples/NoModules/Chapter 14/Ex14_01A/Box.h [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+  {
+    return length*width*height;
+  }
+
+  // Accessors
+  double getLength() const { return length; }
+  double getWidth()  const { return width; }
+  double getHeight() const { return height; }
+
+  friend std::ostream& operator<<(std::ostream& stream, const Box& box);
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/NoModules/Chapter 14/Ex14_01A/Box.h]: Old code...

File: Examples/NoModules/Chapter 14/Ex14_02/Box.h [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+  double getHeight() const { return m_height; }
+
+protected:                   // Protected to facilitate further examples
+  double m_length {1.0};     // later this chapter (should normally be private)
+  double m_width {1.0};
+  double m_height {1.0};
+};
+
+// Stream output for Box objects
+inline std::ostream& operator<<(std::ostream& stream, const Box& box)
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/NoModules/Chapter 14/Ex14_02/Box.h:34]: Yeah, I also noticed. It's on my list of errata...

File: Examples/NoModules/Chapter 14/Ex14_01A/Box.h [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+};
+
+// Stream output for Box objects
+inline std::ostream& operator<<(std::ostream& stream, const Box& box)
+{
+  stream << " Box(" 
+    << std::setw(2) << box.length << ','
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/NoModules/Chapter 14/Ex14_01A/Box.h]: Not really...

File: Examples/NoModules/Chapter 14/Ex14_01A/Carton.h [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+#ifndef CARTON_H
+#define CARTON_H
+#include <string>
+#include <string_view>
+#include "Box.h"
+
+class Carton : private Box
+{
+private:
+  std::string material;
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/NoModules/Chapter 14/Ex14_01A/Carton.h]: Old code...

File: Examples/NoModules/Chapter 14/Ex14_03/Carton.h [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+  { std::cout << "Carton(string_view) called.\n"; }
+
+  Carton(double side, std::string_view material) : Box{side}, m_material{material}
+  { std::cout << "Carton(double,string_view) called.\n"; }
+
+  Carton(double l, double w, double h, std::string_view material)
+    : Box{l, w, h}, m_material{material}
+    { std::cout << "Carton(double,double,double,string_view) called.\n"; }  
+
+  // Copy constructor (wrong)
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/NoModules/Chapter 14/Ex14_03/Carton.h:24]: The text starts with the wrong version, which it explicitly states won't work. Idea is that the reader can verify the (wrong) output, then fix it as instructed in the text, and then verify again.

File: Examples/NoModules/Chapter 14/Ex14_07/FoodContainer.h [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+
+  FoodContainer(std::string_view name, double density, double volume)
+    : name {name}, density {density}, volume {volume}
+  { std::cout << "FoodContainer(string_view,double,double) called.\n"; }
+
+  ~FoodContainer() { std::cout << "FoodContainer destructor" << std::endl; }
+
+  double getWeight() const { return volume * density; }
+
+protected:  // Protected to make initialization in Carton constructor word (book says private)
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/NoModules/Chapter 14/Ex14_07/FoodContainer.h:24]: Yup

File: Examples/NoModules/Chapter 14/Ex14_07A/FoodContainer.h [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+
+  FoodContainer(std::string_view name, double density, double volume)
+    : name {name}, density {density}, volume {volume}
+  { std::cout << "FoodContainer(string_view,double,double) called.\n"; }
+
+  ~FoodContainer() { std::cout << "FoodContainer destructor" << std::endl; }
+
+  double getWeight() const { return volume * density; }
+
+protected:  // Protected to make initialization in Carton constructor word (book says private)
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/NoModules/Chapter 14/Ex14_07A/FoodContainer.h:24]: Yup. And I also meant work, not word.

File: Exercises/NoModules/Chapter 14/Soln14_01/Animals.cpp (hunk 1 above)
Comment (Peter-VanWeert) [reply Exercises/NoModules/Chapter 14/Soln14_01/Animals.cpp:14]: Meh...

**Pull request merged**